File: src/main/java/cz/vhromada/catalog/web/converter/MovieMapper.java
package cz.vhromada.catalog.web.converter;

import cz.vhromada.catalog.entity.Movie;
import cz.vhromada.catalog.web.fo.MovieFO;
import cz.vhromada.converter.orika.MapperConfig;

import ma.glasnost.orika.MapperFactory;
import org.springframework.stereotype.Component;

/**
 * A class represents mapper configuration for movie.
 *
 * @author Vladimir Hromada
 */
@Component("movieMapper")
public class MovieMapper implements MapperConfig {

    @Override
    public void config(final MapperFactory mapperFactory) {
        mapperFactory.classMap(MovieFO.class, Movie.class)
                .field("media{}", "media{length}")
                .exclude("imdbCode")
                .byDefault()
                .customize(new MovieCustomMapper())
                .register();
    }

}

File: src/main/kotlin/cz/vhromada/catalog/web/mapper/MovieMapper.kt
package cz.vhromada.catalog.web.mapper

import cz.vhromada.catalog.entity.Movie
import cz.vhromada.catalog.web.fo.MovieFO

/**
 * An interface represents mapper for movies.
 *
 * @author Vladimir Hromada
 */
interface MovieMapper {

    /**
     * Returns FO for movie.
     *
     * @param source movie
     * @return FO for movie
     */
    fun map(source: Movie): MovieFO

    /**
     * Returns movie.
     *
     * @param source FO for movie
     * @return movie
     */
    fun mapBack(source: MovieFO): Movie

}
